homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,117 @@
+                local_schema = vol.Schema(
+                    {vol.Required(CONF_HOST, default=user_input[CONF_HOST]): str}
+                )
+
+        if user_input is not None:
+            try:
+                serial = await validate_host_input(user_input[CONF_HOST])
+
+                await self.async_set_unique_id(serial)
+                # check if found before
+                self._abort_if_unique_id_configured(
+                    updates={
+                        CONF_HOST: self._discovered_host,
+                    }
+                )
+
+                self._config_context[CONF_HOST] = user_input[CONF_HOST]
+
+                return self.async_create_entry(
+                    title="Fireplace", data=self._config_context
+                )
+
+            except (ConnectionError, ClientConnectionError):
+                errors["base"] = "cannot_connect"
+
+                return self.async_show_form(
+                    step_id="local_config",
+                    errors=errors,
+                    description_placeholders={CONF_HOST: user_input[CONF_HOST]},
+                    data_schema=vol.Schema(
+                        {vol.Required(CONF_HOST, default=user_input[CONF_HOST]): str}
+                    ),
+                )
+
+        return self.async_show_form(
+            step_id="local_config", errors=errors, data_schema=local_schema
+        )
+
+    async def async_step_pick_device(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
-        """Handle the initial step."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
-            )
-        errors = {}
+        """Pick which device to configure."""
+
+        if user_input is not None:
+            return await self.async_step_local_config(user_input=user_input)
 
-        try:
-            serial = await validate_input(self.hass, user_input[CONF_HOST])
-        except (ConnectionError, ClientConnectionError):
-            errors["base"] = "cannot_connect"
-        else:
-            await self.async_set_unique_id(serial)
-            self._abort_if_unique_id_configured(
-                updates={CONF_HOST: user_input[CONF_HOST]}
-            )
-
-            return self.async_create_entry(
-                title="Fireplace",
-                data={CONF_HOST: user_input[CONF_HOST]},
-            )
         return self.async_show_form(
-            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            step_id="pick_device",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_HOST): vol.In(self._discovered_hosts)}
+            ),
         )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Start the user flow."""
+
+        # If the integration was not triggered by DHCP attempt a quick local discovery
```

Comment:
There is no dhcp discovery here?

Reply:
No but there will be in the next PR -> I'll pull it out